services/api/main.go

```
@@ -119,15 +122,29 @@ func main() {
 	//tracer = otel.Tracer(name)
 	//meter = otel.Meter(name)
 	//logger = otelslog.NewLogger(name) // Replace with actual logger initialization
+
+	nc, err := natutils.GetNatsConn()
+	defer func(nc *nats.Conn) {
+		err := nc.Drain()
+		if err != nil {
+		}
+	}(nc)
+	kv, idempotentCtx, err := natutils.NewKeyValueStore(jetstream.KeyValueConfig{Bucket: "idempotent_requests", TTL: time.Hour * 24})
+	if err != nil {
+		println(err.Error(), "error getting new key value store")
+		return
+	}
+
 	e := echo.New()
 
 	// Middleware
+	e.Use(appmiddleware.IdempotencyMiddleware(kv, idempotentCtx))
```

This should come after these as they count the number of requests for load shedding and augments them
```
e.Use(appmiddleware.RequestCounterMiddleware)
e.Use(appmiddleware.AddRouteToCTXMiddleware) 
	// If load is too high, fail before we process anything else. this may need to be moved after logging
	e.Use(echo.WrapMiddleware(loadshedhttp.NewHandlerMiddleware(appmiddleware.CreateShedder(), loadshedhttp.HandlerOptionCallback(&appmiddleware.RejectionHandler{}))))
```

This way even if the request is idempotent if the server is overloaded we can fail fast and shed. 
From my research we want to follow:
- load shedding
- rate limiting
- idempotency
- other middleware
- app logic